pkg/spec/populators_test.go

```diff
@@ -601,6 +604,236 @@ func TestPopulateEnvFrom(t *testing.T) {
 	}
 }
 
+func TestPopulateServicePortNames(t *testing.T) {
+	serviceName := "batman"
+	tests := []struct {
+		name               string
+		inputServicePorts  []api_v1.ServicePort
+		outputServicePorts []api_v1.ServicePort
+	}{
+		{
+			name: "Passing only one servicePort, no population should happen",
+			inputServicePorts: []api_v1.ServicePort{
+				{
+					Port: 8080,
+				},
+			},
+			outputServicePorts: []api_v1.ServicePort{
+				{
+					Port: 8080,
+				},
+			},
+		},
+		{
+			name: "Passing multiple servicePorts with no names, population should happen",
+			inputServicePorts: []api_v1.ServicePort{
+				{
+					Port: 8080,
+				},
+				{
+					Port: 8081,
+				},
+			},
+			outputServicePorts: []api_v1.ServicePort{
+				{
+					Name: fmt.Sprintf("%v-%v", serviceName, 8080),
+					Port: 8080,
+				},
+				{
+					Name: fmt.Sprintf("%v-%v", serviceName, 8081),
+					Port: 8081,
+				},
+			},
+		},
+		{
+			name: "Passing multiple servicePorts with names, no population should happen",
+			inputServicePorts: []api_v1.ServicePort{
+				{
+					Name: fmt.Sprintf("%v-%v", "prepopulated", 8080),
+					Port: 8080,
+				},
+				{
+					Name: fmt.Sprintf("%v-%v", "prepopulated", 8081),
+					Port: 8081,
+				},
+			},
+			outputServicePorts: []api_v1.ServicePort{
+				{
+					Name: fmt.Sprintf("%v-%v", "prepopulated", 8080),
+					Port: 8080,
+				},
+				{
+					Name: fmt.Sprintf("%v-%v", "prepopulated", 8081),
+					Port: 8081,
+				},
+			},
+		},
+		{
+			name: "Passing multiple servicePorts, some with names, some without, selective population should happen",
+			inputServicePorts: []api_v1.ServicePort{
+				{
+					Name: fmt.Sprintf("%v-%v", "prepopulated", 8080),
+					Port: 8080,
+				},
+				{
+					Name: fmt.Sprintf("%v-%v", "prepopulated", 8081),
+					Port: 8081,
+				},
+				{
+					Port: 8082,
+				},
+				{
+					Port: 8083,
+				},
+			},
+			outputServicePorts: []api_v1.ServicePort{
+				{
+					Name: fmt.Sprintf("%v-%v", "prepopulated", 8080),
+					Port: 8080,
+				},
+				{
+					Name: fmt.Sprintf("%v-%v", "prepopulated", 8081),
+					Port: 8081,
+				},
+				{
+					Name: fmt.Sprintf("%v-%v", serviceName, 8082),
+					Port: 8082,
+				},
+				{
+					Name: fmt.Sprintf("%v-%v", serviceName, 8083),
+					Port: 8083,
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			populateServicePortNames(serviceName, test.inputServicePorts)
+
+			if !reflect.DeepEqual(test.inputServicePorts, test.outputServicePorts) {
+				t.Errorf("For input\n%v\nExpected output to be\n%v", spew.Sprint(test.inputServicePorts), spew.Sprint(test.outputServicePorts))
+			}
+		})
+	}
+}
+
+func TestParsePortMapping(t *testing.T) {
```

Comment:
>the placement of this function is wrong can, you place it in `resources.go` or `util.go` ? Because this is file of all the `populators` !

Since the code will move so should the tests!

Reply:
moved